### ISSUE TITLE
AutoPricer seemingly broken?

### ISSUE THREAD

**Issue by JellyTale** [Unovamata/Neopets-Shop-And-Attic-Autobuyer-Cracked#49]
When pulling my shop data in, and clicking on "Start AutoPricing," my console spits out the following:

```Error handling response: TypeError: Cannot read properties of undefined (reading 'some')
    at chrome-extension://loacjeoahhapmaadcnoajpmnolgambcm/src/options/Autopricer/autopricer.js:332:47
    at NodeList.forEach (<anonymous>)
    at chrome-extension://loacjeoahhapmaadcnoajpmnolgambcm/src/options/Autopricer/autopricer.js:321:26
    at chrome-extension://loacjeoahhapmaadcnoajpmnolgambcm/src/common/utils.js:462:13```

This is with the default config and items that are all in the JN database.

**Comment by JellyTale** [comment]
A similar error is thrown when selecting "Update Shop Prices" on the actual store page.

**Comment by Unovamata** [comment]
Hi JellyTale! Sorry for the late reply. By any chance have you clicked the "Reset All Settings" button at the bottom of any the extension pages?

If you could send me another console log it would be quite helpful as the one you attached does not tells me outright the problem you are experiencing. Please head to the [FAQs](https://github.com/Unovamata/Neopets-Shop-And-Attic-Autobuyer-Cracked/wiki/FAQsrl) page for a rundown on how to attach errors to Github's issues effectively.

Thank you very much for your cooperation! I will be waiting patiently for your reply.

I hope you continue to use NeoBuyer+ in the foreseeable future.

**Comment by JellyTale** [comment]
Hiya! I'm so sorry - this was right before I went to bed, so I missed out the actual TypeError (facepalm)

![image](https://github.com/Unovamata/Neopets-Shop-And-Attic-Autobuyer-Cracked/assets/6509290/9f9bc544-0de7-41be-b87b-f953261a6aca)

I did reset my settings afterwards, as I figured it may have been an issue that I had created, but this had happened prior to that as well.

**Comment by Unovamata** [comment]
Do not worry! Mistakes happen. 😆

By any chance can you click on the URLs in the console and send me a screenshot of those lines as well? It would be super helpful, sorry for asking for so much information. I need to be a thorough as possible to solve the error.

Did resetting your settings fixed the issue?
What browser were you using while AutoPricing?

This may be caused by the way NeoBuyer+ initializes variables on its first run but it can also be a browser specific problem. Regardless, I will wait for your reply. Thank you very much for your time.

**Comment by JellyTale** [comment]
Hi again! So, these are shared in the same order that the issues are thrown:

![image](https://github.com/Unovamata/Neopets-Shop-And-Attic-Autobuyer-Cracked/assets/6509290/bd75871c-bc75-48de-879a-922b0a4768cc)


![image](https://github.com/Unovamata/Neopets-Shop-And-Attic-Autobuyer-Cracked/assets/6509290/2042eecb-c37a-4748-bb6e-586b213199b3)

![image](https://github.com/Unovamata/Neopets-Shop-And-Attic-Autobuyer-Cracked/assets/6509290/54ee04f3-e95b-4b51-b189-526b3d703c92)


I thought maybe it was running into an undefined blacklist array because... well, I never defined a blacklist, but A. That's not it, at least from what I can tell, and B. Javascript is the bane of my existence with its weird ways of defining vars, and so I will leave that up to you 😆

**Comment by Unovamata** [comment]
Awesome! You're right! This error is happening because you did not defined a blacklist for AutoPricing. I forgot to add that failsafe.

You can always use the base blacklist:

Forgotten Shore Map Piece
Petpet Laboratory Map
Piece of a treasure map
Secret Laboratory Map
Space Map
Spooky Treasure Map
Underwater Map Piece

I will fix this in the last 0.7V, for now you can use that blacklist and deactivate it when the next version releases. Thank you so much!
